### PR TITLE
Add example code for Performance threshold alert

### PR DESCRIPTION
**Pull request by jfcong**
Add example code for Performance threshold alert for 2nd gen function alert to discord.

### CODE REVIEW

File: 2nd-gen/alerts-to-discord/functions/index.js
Comment (kaibolay) [inline 2nd-gen/alerts-to-discord/functions/index.js:189]: Do you think we should pull this code (probably also the `try`/`catch`) into `postMessageToDiscord()`? It's not replicated three times with only minimal differences in the info/error message being logged. We could just make the error messages more generic.

The complication is that this error handling is currently in the snippets (`[START v2SendToDiscord]`/`[END v2SendToDiscord]`)...

Comment (jfcong) [reply 2nd-gen/alerts-to-discord/functions/index.js:189]: I discussed this with Victoria, I think what we can do is leave this as is, but in the documentation, just link to the code snippet without showing the code itself, so it doesn't look as redundant in the doc. wdyt?

Ideally we can clean this up in the code snippets as well but as you said, it's currently used in the doc...

File: 2nd-gen/alerts-to-discord/functions/index.js
Comment (jfcong) [reply 2nd-gen/alerts-to-discord/functions/index.js]: Thanks for the catch! Updated the tag. (ideally we can update it for crash too...)

**Comment by jhuleatt** [comment]
Thanks for the performance sample @jfcong! I'm a bit confused though - how does this PR relate to https://github.com/firebase/functions-samples/pull/1012? Seems like both add the iOS app distro sample. Does it replace that one?

File: 2nd-gen/alerts-to-discord/functions/index.js
Comment (kaibolay) [reply 2nd-gen/alerts-to-discord/functions/index.js:189]: Ok. We can always to the de-duplication in a follow-up.

**Comment by kaibolay** [comment]
> Thanks for the performance sample @jfcong! I'm a bit confused though - how does this PR relate to #1012? Seems like both add the iOS app distro sample. Does it replace that one?

@jhuleatt Jenny's PR builds on top of mine (#1012), so it also includes those changes.

You can look at [only her changes](https://github.com/firebase/functions-samples/pull/1012/files/1b2d4eefa86bf11010927240981c04f7b1ffad66..f7ca5537cf72d23d54895efe82d474ea3f980ef5) by selecting only her commits for the diff.

The plan is that I'll first merge my PR, then Jenny will rebase her branch (so it only includes her changes), update this PR, and then merge it.

**Comment by jfcong** [comment]
@jhuleatt Like Kai said, I will rebase once Kai's PR is submitted :)

Thanks Kai for replying!

**Review by kaibolay — APPROVED** [review]

**Review by jhuleatt — APPROVED** [review]
LGTM, please get tech writer approval of the includecode snippets before merging. Thanks again for contributing a sample!

File: 2nd-gen/alerts-to-discord/functions/index.js
Comment (jhuleatt) [inline 2nd-gen/alerts-to-discord/functions/index.js:176]: nit: consider making this conditional instead of having the "if applicable" qualifier

File: 2nd-gen/alerts-to-discord/functions/index.js
Comment (jhuleatt) [inline 2nd-gen/alerts-to-discord/functions/index.js:178]: This seems like the most important thing - consider moving it higher up in the message

Comment (jfcong) [reply 2nd-gen/alerts-to-discord/functions/index.js:178]: I grouped it by two clusters, the first cluster is alert condition, and the second cluster is the alert violation. This resembles how our alert email displays the information (with a focus on the alert condition as well). I'll let a tech writer provide some input for this one :)

File: 2nd-gen/alerts-to-discord/functions/index.js
Comment (jfcong) [reply 2nd-gen/alerts-to-discord/functions/index.js:176]: I am wondering what you mean by conditional, do you mean having 4 if statements and depending on which one is true, having a different string message? I am leaning towards just keeping it simple because this is just an example for developers and they should be able to format the message however they want.

File: 2nd-gen/alerts-to-discord/functions/index.js
Comment (VictoriaAuto) [inline 2nd-gen/alerts-to-discord/functions/index.js:147]: Update product name to "Firebase Performance Monitoring"

File: 2nd-gen/alerts-to-discord/functions/index.js
Comment (VictoriaAuto) [reply 2nd-gen/alerts-to-discord/functions/index.js:178]: To provide a consistent experience across surfaces, I recommend keeping the current flow. Also, the example isn't that long, and a violation is implied in the description; I think developers will expect that it's a part of the example and will look for it.

Comment (jfcong) [reply 2nd-gen/alerts-to-discord/functions/index.js:178]: sounds good, I'll keep it as is

File: 2nd-gen/alerts-to-discord/functions/index.js
Comment (jfcong) [reply 2nd-gen/alerts-to-discord/functions/index.js]: Updated.

**Review by VictoriaAuto — APPROVED** [review]

File: 2nd-gen/alerts-to-discord/functions/index.js
Comment (jfcong) [reply 2nd-gen/alerts-to-discord/functions/index.js:176]: I am going to keep this simple and not do conditional, since this is meant to just be some sample code

**Pull request merged**
